src/mig/agent/agent.go

```diff
@@ -163,8 +163,8 @@ func runModuleDirectly(mode string, args []byte, pretty bool) (out string) {
 		return fmt.Sprintf(`{"errors": ["module '%s' is not available"]}`, mode)
 	}
 	// instanciate and call module
-	modRunner := modules.Available[mode].Runner()
-	out = modRunner.(modules.Moduler).Run(os.Stdin)
+	run := modules.Available[mode].NewRun()
```

Comment:
If we're going to rely on `NewRun()` being present, it needs to be part of the `modules.Runner` interface to break compilation if a module forgets to implement it.

```go
type Runner interface {
 	Run(io.Reader) string
 	NewRun() interface{}
 	ValidateParameters() error
 }
```